supervision/detection/line_zone.py

```diff
@@ -158,28 +158,29 @@ def trigger(self, detections: Detections) -> Tuple[np.ndarray, np.ndarray]:
             ]
         )
 
+        cross_products_1 = self._cross_product(all_anchors, self.limits[0])
+        cross_products_2 = self._cross_product(all_anchors, self.limits[1])
+        # anchor is in limits if it's on the same side of both limit vectors
+        in_limits = ~np.logical_xor(cross_products_1 > 0, cross_products_2 > 0)
+        # Reduce array to find out if all anchors for a detection are within limits
+        in_limits = np.min(in_limits, axis=0)
+
+        # Calculate which anchors lie to the left of the line
+        triggers = self._cross_product(all_anchors, self.vector) < 0
+        # Reduce to find out if all anchors for a
+        # detection lie to the left (or right) of the line
+        max_triggers = np.max(triggers, axis=0)
```

Comment:
@tc360950 what do you think about calculating this slightly differently, but at the same time naming variables in a more descriptive way:

```python
has_any_left_trigger = np.any(triggers, axis=1)
has_any_right_trigger = np.any(~triggers, axis=1)
is_uniformly_triggered = ~(has_any_left_trigger & has_any_right_trigger)
```

I don't have a particularly strong opinion on this. If you think it's better to stick with your names, we can merge.

Reply:
@SkalskiP I think it makes it much clearer what's going on. I've added this change.